### ISSUE TITLE
Mutable default arguments

### ISSUE THREAD

**Issue by MatthewScholefield** [MycroftAI/mycroft-core#1227]
Throughout the code there are places with default arguments of things like `{}` and `[]`. This is misleading and should be changed to `param=None): param=param or {}`.

Problem:
```Python
def tst(var=[]):

    var += [0]
    print(var)
```

Fixed:
```Python
def tst(var=None):
    var=var or []

    var += [0]
    print(var)
```

The reason is, with the first one, `tst(); tst(); tst()` prints `[0] [0,0] [0,0,0]`.

**Comment by gauravMann** [comment]
Can I work on it? I have to replace every definition by the fixed one right ?

**Comment by MatthewScholefield** [comment]
@gauravMann Feel free to fix everything in one go and when you are finisheed, submit a pull request. If anything doesn't look right we'll notice it in the review process :). Also, not sure if this is what you are talking about, but that `tst()` thing is just showing why it's not good to have mutable default arguments.

**Comment by forslund** [comment]
Resolved by #1236. Thanks @gauravMann !
